### PR TITLE
Enabling cla functionality in mungegithub for contrib.

### PR DESCRIPTION
**Pull request by foxish**
This PR contains:
- Modifications to the CLA munger to enable ping notifications
- Fixes to the deployment, and addition of a new parameter required by the CLA munger.
- Turning on the mungegithub handling of CLA on contrib.

cc @sarahnovotny @bgrant0607

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1774)

### CODE REVIEW

File: mungegithub/mungers/cla.go [hunk 1]
```diff
@@ -37,54 +38,65 @@ Once you've signed, please reply here (e.g. "I signed it!") and we'll verify.  T
...
+	maxPings   = 3
 )
 
 // ClaMunger will check the CLA status of the PR and apply a label.
 type ClaMunger struct {
 	CLAStatusContext string
+	pinger           *c.Pinger
 }
 
+var _ Munger = &ClaMunger{}
```
Comment (eparis) [inline mungegithub/mungers/cla.go]: why?

**Review by eparis — COMMENTED** [review]
Not your PR, but why do we even bother turning the status into a label? I know the old google CLA thing did that, but it seems to serve no useful purpose...

File: mungegithub/mungers/cla.go (hunk 1 above)
Comment (foxish) [inline mungegithub/mungers/cla.go]: Just for the compile-time check that we satisfy the interface definition.

File: mungegithub/mungers/cla.go [hunk 2]
```diff
@@ -113,14 +125,27 @@ func (c *ClaMunger) Munge(obj *githubhelper.MungeObject) {
...
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error(err)
 		return
 	}
+	startDate := c.LastComment(comments, c.MungerNotificationName(claNagNotifyName), nil)
+	who := mungerutil.GetIssueUsers(obj.Issue).Author.Mention().Join()
+
+	// Get a notification if it's time to ping.
+	notif := cla.pinger.SetTimePeriod(timePeriod).SetMaxCount(maxPings).PingNotification(
```
Comment (apelisse) [inline mungegithub/mungers/cla.go]: You could set `TimePeriod` and `MaxCount` once and for all at the beginning

File: mungegithub/mungers/cla.go [hunk 3]
```diff
@@ -113,14 +125,27 @@ func (c *ClaMunger) Munge(obj *githubhelper.MungeObject) {
...
 
-	if obj.HasLabel(cncfClaNoLabel) {
-		// status reported error/failure and we've already applied 'cncf-cla: no' label.
+	// If we are here, that means that the context is failure/error.
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error(err)
 		return
 	}
+	startDate := c.LastComment(comments, c.MungerNotificationName(claNagNotifyName), nil)
```
Comment (apelisse) [inline mungegithub/mungers/cla.go]: I think that doesn't do what you want (or I mean, this should is done automatically)

**Comment by foxish** [comment]
@eparis It might be useful as a way to filter PRs by CLA status, and for the cla:human-approved thing to work. The submit-queue honors the labels and not the status. Do you think there is a way around that?

File: mungegithub/mungers/cla.go (hunk 3 above)
Comment (foxish) [inline mungegithub/mungers/cla.go]: Fixed

**Comment by apelisse** [comment]
Cool thanks! looks good to me!

**Pull request merged**